src/hist/plot.py

```diff
@@ -214,185 +262,408 @@ def plot2d_full(
     return main_art, top_art, side_art
 
 
+def _construct_gaussian_callable(
+    _hist: hist.BaseHist,
+) -> Callable[[np.ndarray, float, float, float], np.ndarray]:
+    x_values = _hist.axes[0].centers
+    hist_values = _hist.values()
+
+    # gaussian with reasonable initial guesses for parameters
+    constant = float(hist_values.max())
+    mean = (hist_values * x_values).sum() / hist_values.sum()
+    sigma = (hist_values * np.square(x_values - mean)).sum() / hist_values.sum()
+
+    # gauss is a closure that will get evaluated in _fit_callable_to_hist
+    def gauss(
+        x: np.ndarray,
+        constant: float = constant,
+        mean: float = mean,
+        sigma: float = sigma,
+    ) -> np.ndarray:
+        # Note: Force np.ndarray type as numpy ufuncs have type "Any"
+        ret: np.ndarray = constant * np.exp(
+            -np.square(x - mean) / (2 * np.square(sigma))
+        )
+        return ret
+
+    return gauss
+
+
+def _fit_callable_to_hist(
+    model: Callable[[np.ndarray], np.ndarray],
+    _hist: hist.BaseHist,
+    likelihood: bool = False,
+) -> "Tuple[np.ndarray, np.ndarray, np.ndarray, Tuple[Tuple[float, ...], np.ndarray]]":
+    """
+    Fit a model, a callable function, to the histogram values.
+    """
+    variances = _hist.variances()
+    if variances is None:
+        raise RuntimeError(
+            "Cannot compute from a variance-less histogram, try a Weight storage"
+        )
+    hist_uncert = np.sqrt(variances)
+
+    # Infer best fit model parameters and covariance matrix
+    xdata = _hist.axes[0].centers
+    popt, pcov = _curve_fit_wrapper(
+        model, xdata, _hist.values(), hist_uncert, likelihood=likelihood
+    )
+    model_values = model(xdata, *popt)
+
+    if np.isfinite(pcov).all():
+        n_samples = 100
+        vopts = np.random.multivariate_normal(popt, pcov, n_samples)
+        sampled_ydata = np.vstack([model(xdata, *vopt).T for vopt in vopts])
+        model_uncert = np.nanstd(sampled_ydata, axis=0)
+    else:
+        model_uncert = np.zeros_like(hist_uncert)
+
+    return model_values, model_uncert, hist_uncert, (popt, pcov)
+
+
+def _plot_fit_result(
+    _hist: hist.BaseHist,
```

Comment:
By the way, if you want a "position only" variable before Python 3.8, you should use `__hist` - mypy literally will consider this a position only variable; it's a convention used in typeshed for the standard library. Though in this case, `histogram` would probably be fine.

Reply:
Thanks &mdash; I didn't know this! Changed this over for all cases where I was using `_hist` to avoid using `hist`.